Frameworks/CoreFoundation/Base.subproj/CFInternal.h

```diff
@@ -696,28 +706,23 @@ CF_INLINE bool __CF_IsBridgedObject(CFTypeRef obj) {
         return true;
     }
 
-    for (unsigned int i = 0; i < __CFRuntimeClassTableSize; i++) {
-        if ((__CFRuntimeObjCClassTable[i] != 0) && 
-            ((object->_cfisa == __CFRuntimeObjCClassTable[i]) ||
-            ([object isKindOfClass:(Class)(__CFRuntimeObjCClassTable[i])]))) {
-            return true;
-        }
-    }
-
-    return false;
+    // Check if the object has the 'bridged object' protocol
+    // NOTE: because our class_conformsToProtocol() checks based only on name, 
+    // it is possible to spoof a bridged object by stating an object conforms to a protocol with the same name.
+    // However, this would not grant access to anything that could not be accessed by more conventional means
+    // (ie: overriding an existing CF/Foundation class)
+    // and as such would not constitute an escalation of privilege.
+    return [object conformsToProtocol:__CFRuntimeGetBridgeProtocol()];

```

Comment:
Prefer `class_conformsToProtocol(object->_cfisa, ...)` -- that will remove Objective-C from purely C++ CFInternal.h consumers.